server/server.go

```diff
@@ -32,10 +32,49 @@ import (
 
 const (
 	pairTimeout = 24 * time.Hour
-	sendTimeout = time.Second
-	recvTimeout = 5 * time.Second
+	prTimeout   = 5 * time.Second
+	keTimeout   = 5 * time.Second
 )
 
+type timer struct {
+	*time.Timer
+}
+
+func (t *timer) after(d time.Duration) <-chan time.Time {
+	if t.Timer == nil {
+		t.Timer = time.NewTimer(d)
+	} else {
+		t.Timer.Reset(d)
```

Comment:
Don't you need to stop and drain the timer if it hasn't fired yet?

```go
if !t.Stop() {
	<-t.C
}
t.Reset(d)
```

Reply:
you're right, let me see what to do about that.